src/main/java/views/add_edit_category_views/AddCategoryV.java

```diff
@@ -48,22 +52,23 @@ public AddCategoryV(CategoryC controller,int monthID, SessionStorage curr_sessio
         panel.add(submit);
         frame.add(panell, BorderLayout.SOUTH);
         panell.add(submit);
-        /**
-         * Attempts to pass in the category information to the controller.
-         * Success message with category name is produced when successful.
-         * Fail message with type of error when failed.
-         */
+
         this.controller = controller;
-        try{
-            controller.addCategoryInMonth(name_input.getText(), String.valueOf(budget_input), monthID, curr_session);
-            JOptionPane.showMessageDialog( this, name_input.getText());
-        }catch (Exception e){
-            JOptionPane.showMessageDialog( this, e.getMessage());
-        }
+        this.monthID = monthID;
+        this.curr_session = curr_session;
+
+        submit.addActionListener(this);
     }
     @Override
-    public void actionPerformed(ActionEvent e) {System.out.println("Click " + e.getActionCommand());
-        System.out.println("Click " + e.getActionCommand());
+    public void actionPerformed(ActionEvent evt) {
+        CategoryOD m = null;
+        try {
+            m = controller.addCategoryInMonth(name_input.getText(), String.valueOf(budget_input), monthID, curr_session);
+        } catch (EntityException e) {
+            JOptionPane.showMessageDialog( this, "This month does not exist in current session. Please go to add month page.");
+        }
+        assert m != null;
```

Comment:
Maybe you need to use if instead of assert?

Reply:
Or move the line below that uses the m variable to the inside of the try branch